tests/drop_counters/test_drop_counters.py

```diff
@@ -429,3 +440,152 @@ def test_dst_ip_absent(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields):
+            ip_src=ip_class_e,
+            ip_dst=pkt_fields["ipv4_dst"], # VM source
+            tcp_sport=pkt_fields["tcp_sport"],
+            tcp_dport=pkt_fields["tcp_dport"])
+
+        base_verification("L3", pkt, ptfadapter, duthost, setup["combined_drop_counter"], ptf_tx_port_id, tx_dut_ports[dut_iface])
+
+        # Verify packets were not egresed the DUT
+        exp_pkt = expected_packet_mask(pkt)
+        testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["neighbor_sniff_ports"])
+
+
+@pytest.mark.parametrize("addr_type, addr_direction", [("ipv4", "src"), ("ipv6", "src"), ("ipv4", "dst"),
+                                                        ("ipv6", "dst")])
+def test_ip_is_zero_addr(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields, addr_type, addr_direction):
+    """
+    @summary: Verify that packet with "0.0.0.0" source or destination IP address is dropped and L3 drop counter incremented
+    """
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+    zero_ipv4 = "0.0.0.0"
+    zero_ipv6 = "::0"
+
+    pkt_params = {
+        "eth_dst": dst_mac, # DUT port
+        "eth_src": src_mac, # PTF port
+        "tcp_sport": pkt_fields["tcp_sport"],
+        "tcp_dport": pkt_fields["tcp_dport"]
+        }
+
+    if addr_type == "ipv4":
+        if addr_direction == "src":
+            pkt_params["ip_src"] = zero_ipv4
+            pkt_params["ip_dst"] = pkt_fields["ipv4_dst"] # VM source
+        elif addr_direction == "dst":
+            pkt_params["ip_src"] = pkt_fields["ipv4_src"] # VM source
+            pkt_params["ip_dst"] = zero_ipv4
+        else:
+            pytest.fail("Incorrect value specified for 'addr_direction'. Supported parameters: 'src' and 'dst'")
+        pkt = testutils.simple_tcp_packet(**pkt_params)
+    elif addr_type == "ipv6":
+        if not pkt_fields["ipv6_dst"]:
+            pytest.skip("BGP neighbour with IPv6 addr was not found")
+        if addr_direction == "src":
+            pkt_params["ipv6_src"] = zero_ipv6
+            pkt_params["ipv6_dst"] = pkt_fields["ipv6_dst"] # VM source
+        elif addr_direction == "dst":
+            pkt_params["ipv6_src"] = pkt_fields["ipv6_src"] # VM source
+            pkt_params["ipv6_dst"] = zero_ipv6
+        else:
+            pytest.fail("Incorrect value specified for 'addr_direction'. Supported parameters: 'src' and 'dst'")
+        pkt = testutils.simple_tcpv6_packet(**pkt_params)
+    else:
+        pytest.fail("Incorrect value specified for 'addr_type' test parameter. Supported parameters: 'ipv4' or 'ipv6'")
+
+    logger.info(pkt_params)
+    base_verification("L3", pkt, ptfadapter, duthost, setup["combined_drop_counter"], ptf_tx_port_id,
+                        tx_dut_ports[dut_iface])
+
+    # Verify packets were not egresed the DUT
+    exp_pkt = expected_packet_mask(pkt)
+    testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["neighbor_sniff_ports"])
```

Comment:
In case of dst ip is zero and there is a HW bug allowing such packets to go out from the switch it is not clear why you check only neighbor_sniff_ports? Maybe verify on every port?

Reply:
Fixed, added verification on every port.